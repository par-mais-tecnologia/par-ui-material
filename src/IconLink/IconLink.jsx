import React, { PureComponent } from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import PropTypes from 'prop-types'

class IconLink extends PureComponent {
  render () {
    const { color, icon, onClick, text, fontSize, className } = this.props

    return (
      <Grid container className={`b pointer ${className}`} onClick={onClick}>
        <Typography align='center' variant='body1'>
          <i style={{ color, fontSize: fontSize - 3 }} className={`par-icon-${icon} mr2`} />
          <b style={{ color, fontSize }}>{text}</b>
        </Typography>
      </Grid>
    )
  }
}

IconLink.propTypes = {
  color: PropTypes.string,
  className: PropTypes.string,
  fontSize: PropTypes.number,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string
}

IconLink.defaultProps = {
  color: '#CCC',
  fontSize: 15,
  icon: 'reload',
  text: 'Texto'
}

export default IconLink
