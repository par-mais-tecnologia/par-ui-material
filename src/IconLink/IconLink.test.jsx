import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import IconLink from './'

configure({ adapter: new Adapter() })

const wrapper = shallow(
  <IconLink
    color={'#909090'}
    fontSize={15}
    icon={'reload'}
    onClick={() => console.log('onClick function!')}
    text={'Refazer BIO'}
  />
)

export const test = describe('IconLink', () => {
  it('shallow IconLink', () => {
    expect(wrapper).toBeTruthy()
  })

  it('shallow IconLink match Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
