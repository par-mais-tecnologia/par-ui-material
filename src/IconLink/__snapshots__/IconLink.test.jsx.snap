// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink shallow IconLink match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconLink
    color="#909090"
    fontSize={15}
    icon="reload"
    onClick={[Function]}
    text="Refazer BIO"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Typography)
        align="center"
        variant="body1"
      >
        <i
          className="par-icon-reload mr2"
          style={
            Object {
              "color": "#909090",
              "fontSize": 12,
            }
          }
        />
        <b
          style={
            Object {
              "color": "#909090",
              "fontSize": 15,
            }
          }
        >
          Refazer BIO
        </b>
      </WithStyles(Typography)>,
      "className": "b pointer undefined",
      "container": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": Array [
          <i
            className="par-icon-reload mr2"
            style={
              Object {
                "color": "#909090",
                "fontSize": 12,
              }
            }
          />,
          <b
            style={
              Object {
                "color": "#909090",
                "fontSize": 15,
              }
            }
          >
            Refazer BIO
          </b>,
        ],
        "variant": "body1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "par-icon-reload mr2",
            "style": Object {
              "color": "#909090",
              "fontSize": 12,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Refazer BIO",
            "style": Object {
              "color": "#909090",
              "fontSize": 15,
            },
          },
          "ref": null,
          "rendered": "Refazer BIO",
          "type": "b",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          align="center"
          variant="body1"
        >
          <i
            className="par-icon-reload mr2"
            style={
              Object {
                "color": "#909090",
                "fontSize": 12,
              }
            }
          />
          <b
            style={
              Object {
                "color": "#909090",
                "fontSize": 15,
              }
            }
          >
            Refazer BIO
          </b>
        </WithStyles(Typography)>,
        "className": "b pointer undefined",
        "container": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": Array [
            <i
              className="par-icon-reload mr2"
              style={
                Object {
                  "color": "#909090",
                  "fontSize": 12,
                }
              }
            />,
            <b
              style={
                Object {
                  "color": "#909090",
                  "fontSize": 15,
                }
              }
            >
              Refazer BIO
            </b>,
          ],
          "variant": "body1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "par-icon-reload mr2",
              "style": Object {
                "color": "#909090",
                "fontSize": 12,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Refazer BIO",
              "style": Object {
                "color": "#909090",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "Refazer BIO",
            "type": "b",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
